quinn-proto/src/connection/mod.rs

```diff
@@ -1850,6 +1878,49 @@ impl Connection {
         self.streams.queue_max_stream_id(pending);
     }
 
+    /// Modify the number of open paths allowed when multipath is enabled
+    ///
+    /// When reducing the number of concurrent paths this will only affect delaying sending
+    /// new MAX_PATH_ID frames until fewer than this number of paths are possible.  To
+    /// actively reduce paths they must be closed using [`Connection::close_path`], which
+    /// can also be used to close not-yet-opened paths.
+    ///
+    /// If multipath is not enabled in the [`TransportConfig`] this can not enable mulipath
+    /// and will fail.
+    pub fn set_max_concurrent_paths(
+        &mut self,
+        now: Instant,
+        count: NonZeroU32,
+    ) -> Result<(), MultipathNotNegotiated> {
+        if !self.is_multipath_negotiated() {
+            return Err(MultipathNotNegotiated { _private: () });
+        }
+        self.max_concurrent_paths = count;
+
+        let in_use_count = self.local_max_path_id.0 - self.abandoned_paths.len() as u32;
+        let desired_count = u32::from(count);
+        let extra_needed = desired_count.saturating_sub(in_use_count);
+        let new_max_path_id = self.local_max_path_id.0 + extra_needed;
+
+        self.set_max_path_id(now, PathId(new_max_path_id));
```

Comment:
you can just do `count.get()` also no need to access the content of path id to build a new one, just `saturating_add` to it
```suggestion
        let extra_needed = count.get().saturating_sub(in_use_count);
        let new_max_path_id = self.local_max_path_id.saturating_add(extra_needed);
        
        self.set_max_path_id(now, new_max_path_id);
```

Reply:
TIL about `NonZeroUxx::get`.  And the `PathId::saturating_add/sub`.  That is more ergonomic indeed.